### ISSUE TITLE
Styled components v6 doesn't work with testing library

### ISSUE THREAD

**Issue by TvojTatk0** [styled-components/styled-components#4094]
### System:
 - OS: macOS 13.4.1
 - CPU: (8) arm64 Apple M1 Pro
 - Memory: 856.61 MB / 32.00 GB
 - Shell: 5.9 - /bin/zsh
### Binaries:
 - Node: 18.13.0 - ~/.nvm/versions/node/v18.13.0/bin/node
 - Yarn: 1.22.19 - ~/.nvm/versions/node/v18.13.0/bin/yarn
 - npm: 8.19.3 - ~/.nvm/versions/node/v18.13.0/bin/npm
 - pnpm: 7.25.1 - ~/.nvm/versions/node/v18.13.0/bin/pnpm
### npmPackages:
 - styled-components: ^6.0.4 => 6.0.4 

## Reproduction
I have storybook component library based on:
- react: 18.2.0
- vite: 4.4.4
- vitest: 0.33.0
- @testing-library/jest-dom: 5.16.5
- @testing-library/react: 14.0.0
## Steps to reproduce
1. Write a simple component (Button) with some css properties
2. White a simple test using `toHaveStyle` matcher which should have passed.

## Expected Behavior
Test should passed

## Actual Behavior
Test ends up with error like that (property ButtonText is not defined in whole project).
```
Error: expect(element).toHaveStyle()

- Expected

- background: rgb(66, 190, 101);
- border-color: #42be65;
- color: rgb(255, 255, 255);
- min-height: 3.75rem;
- padding-bottom: 1.125rem;
- padding-top: 1.125rem;
+ color: ButtonText;
 ❯ src/components/button/rectangle/RectangleButton.test.tsx:22:20
     20|     const button = screen.getByText(/Rectangle button/i);
     21| 
     22|     expect(button).toHaveStyle(rectangleButtonStyles);
       |                    ^
     23|   });
     24| 
```

## Additional info
Basically is the same issue as it is described here https://github.com/styled-components/styled-components/issues/3685 where some versions of SC does produce this behavior.

**Comment by ziolekjj** [comment]
Can you make sure you're resolving `styled-components` package in your test runner to the correct bundle (browser/server/native)? My assumption is that you're importing `dist/styled-components.cjs.js`, which is compiled strictly for the server

**Comment by StigoSvK** [comment]
@ziolekjj It seems you are right. 👍 
![image](https://github.com/styled-components/styled-components/assets/28844291/917285cc-bbd4-4324-a8bd-dd2968a5999d)
When we do this, `toHaveStyles` matcher works like before. 

But why `vitest` import wrong bundle by default? We have had no issues with version 5.3.4 and I saw comment where somebody else also revert from version 6.0.0-RC1 to 5.3.3.

**Comment by linuspahl** [comment]
We migrated from `5.3.11` to `6.0.4` and face the same problem. Our setup is different since we use jest. I just want to mention that it looks like both versions import `dist/styled-components.cjs.js`.

**Comment by domarmstrong** [comment]
You can map the import in jest config https://github.com/styled-components/styled-components/issues/4081

**Comment by quantizor** [comment]
Closing — this issue is stale and relates to an older version. If still reproducible on the latest v6.3.x release, please open a new issue with a minimal reproduction.
